src/components/PostDetails.tsx

```diff
@@ -1,160 +1,148 @@
+          </p>
+        </div>
+        <div className="block">
+          {loaded && (
+            <Loader />
+          )}
+        </div>
+      </div>
+    );
   }
 
-  useEffect(loadComments, [post.id]);
+  if (hasError) {
+    return (
+      <div className="content" data-cy="PostDetails">
+        <div className="block">
+          <h2 data-cy="PostTitle">
+            {`#${selectedPost?.id}: ${selectedPost?.title}`}
+          </h2>
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
+          <p data-cy="PostBody">
+            {selectedPost?.body}
+          </p>
+        </div>
+        <div className="block">
+          <div className="notification is-danger" data-cy="CommentsError">
+            Something went wrong
+          </div>
+        </div>
+      </div>
+    );
+  }
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
+  if (comments.length === 0) {
+    return (
+      <div className="content" data-cy="PostDetails">
+        <div className="block">
+          <h2 data-cy="PostTitle">
+            {`#${selectedPost?.id}: ${selectedPost?.title}`}
+          </h2>
 
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
-  };
+          <p data-cy="PostBody">
+            {selectedPost?.body}
+          </p>
+        </div>
+        <div className="block">
+          <p className="title is-4" data-cy="NoCommentsMessage">
+            No comments yet
+          </p>
+        </div>
 
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
-
-  const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
-  };
-
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
+        {!loaded && !hasError && !visible && (
+          <button
+            data-cy="WriteCommentButton"
+            type="button"
+            className="button is-link"
+            onClick={() => setVisible(true)}
```

Comment:
Do not use anonymous function, create separate handlers. Fix all similar places

Reply:
didn't fix it